src/lib/x509/x509_ext.cpp

```diff
@@ -1510,6 +1510,10 @@ void ASBlocks::decode_inner(const std::vector<uint8_t>& in) {
 void ASBlocks::ASIdentifiers::encode_into(Botan::DER_Encoder& into) const {
    into.start_sequence();
 
+   if (!m_asnum.has_value() && !m_rdi.has_value()) {
+      throw Encoding_Error("One of asnum, rdi must be present");
+   }
+
```

Comment:
When just doing `auto ext = std::make_unique<ASBlocks>()`, nothing is set initially, so the (std::nullopt, std::nullopt) case that was previously caught by the `ASIdentifiers` constructor now needs to be caught here too